### PR TITLE
Introduced argument parsing to customize host/port

### PR DESCRIPTION
**Pull request by dbrgn**
Using docopt to parse. There are currently two optional parameters: -p for the port and -i for the ip.

This replaces the `PORT` env variable.

Please review @rnestler @schmijos :)

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -3,20 +3,75 @@
...
+    -p PORT  The port to listen on [default: 3000].
+    -i IP    The ipv4 address to listen on [default: 127.0.0.1].
+";
+
+// We have to create a wrapper because we cannot implement Decodable for types
+// outside this crate (E0117).
+#[derive(Debug)]
+struct Ipv4 {
+    ip: Ipv4Addr,
+}
```
Comment (rnestler) [inline src/main.rs]: Hmm this is a pitty, but isn't there an easier solution? See also https://github.com/rust-lang/rfcs/issues/493 for some more information.

**Comment by rnestler** [comment]
Why do you want to replace the env variable? Why not both?

File: src/main.rs (hunk 1 above)
Comment (dbrgn) [inline src/main.rs]: We could of course change `flag_i` to a string and do manual validation and conversion, but I think implementing `Decodable` is much nicer than that.

The rfc493 discussion didn't really lead anywhere, did it?

**Comment by dbrgn** [comment]
The reason for the env variable was to prevent an argument system. If we had both, which would take precedence? Also, I could imagine new commandline arguments that wouldn't make sense as env variables.

Do you think it's worth it to bring the env var back, and to introduce it for the IP too?

**Comment by dbrgn** [comment]
Rebased and reorganized code.

**Pull request merged**